### PR TITLE
Added preprod and prod NOTM env.

### PR DESCRIPTION
**Pull request by Mjwillis**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,6 @@ terraform.tfstate*
 *.tfplan
 graph.png
 node_modules/
+#IntelliJ specific files
```
Comment (glenjamin) [inline .gitignore]: these should probably just be in your system level gitignore

File: notm/preprod/azure-preprod.tf [hunk 2]
```diff
@@ -0,0 +1,30 @@
+variable "azure_subscription_id" {
```
Comment (glenjamin) [inline notm/preprod/azure-preprod.tf]: this should be a symlink to `azure-prod`, it's intended to be a pointer to which of the two subscriptions we use

File: notm/preprod/github.tf [hunk 3]
```diff
@@ -0,0 +1,4 @@
+provider "github" {
```
Comment (glenjamin) [inline notm/preprod/github.tf]: likewise, symlink

File: notm/preprod/ips.tf [hunk 4]
```diff
@@ -0,0 +1,7 @@
+variable "ips" {
```
Comment (glenjamin) [inline notm/preprod/ips.tf]: symlink

File: notm/preprod/main.tf [hunk 5]
```diff
@@ -0,0 +1,217 @@
+terraform {
+    required_version = ">= 0.9.2"
+    backend "azure" {
+        resource_group_name = "webops-preprod"
```
Comment (glenjamin) [inline notm/preprod/main.tf]: this section should all be prod, and only the `key` param should be project-specific

File: notm/preprod/main.tf [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+    name = "webapp-ssl"
+    resource_group_name = "${azurerm_resource_group.group.name}"
+    deployment_mode = "Incremental"
+    template_body = "${file("../../shared/appservice-ssl.template.json")}"
+
+    parameters {
+        name = "${azurerm_template_deployment.webapp.parameters.name}"
+        hostname = "${azurerm_dns_cname_record.cname.name}.${azurerm_dns_cname_record.cname.zone_name}"
+        keyVaultId = "${azurerm_key_vault.vault.id}"
+        keyVaultCertName = "csraDOTserviceDOThmppsDOTdsdDOTio"
```
Comment (glenjamin) [inline notm/preprod/main.tf]: needs changing, possibly using a string replace function on hostname

File: notm/preprod/main.tf [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+        service = "${var.tags["Service"]}"
+        environment = "${var.tags["Environment"]}"
+    }
+
+    depends_on = ["azurerm_template_deployment.webapp"]
+}
+
+resource "azurerm_dns_cname_record" "cname" {
+    name = "notm-preprod"
+    zone_name = "hmpps.dsd.io"
```
Comment (glenjamin) [inline notm/preprod/main.tf]: production zone is `service.hmpps.dsd.io`

File: notm/prod/azure-prod.tf [hunk 8]
```diff
@@ -0,0 +1,30 @@
+variable "azure_subscription_id" {
```
Comment (glenjamin) [inline notm/prod/azure-prod.tf]: symlink

File: notm/prod/github.tf [hunk 9]
```diff
@@ -0,0 +1,4 @@
+provider "github" {
```
Comment (glenjamin) [inline notm/prod/github.tf]: symlink

File: notm/prod/ips.tf [hunk 10]
```diff
@@ -0,0 +1,8 @@
+variable "ips" {
```
Comment (glenjamin) [inline notm/prod/ips.tf]: symlink

File: notm/preprod/main.tf [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+resource "azurerm_template_deployment" "webapp-whitelist" {
+    name = "webapp-whitelist"
+    resource_group_name = "${azurerm_resource_group.group.name}"
+    deployment_mode = "Incremental"
+    template_body = "${file("../../shared/appservice-whitelist.template.json")}"
+
+    parameters {
+        name = "${azurerm_template_deployment.webapp.parameters.name}"
+        ip1 = "${var.ips["office"]}"
+        ip2 = "${var.ips["quantum"]}"
```
Comment (glenjamin) [inline notm/preprod/main.tf]: should include health kick as well, which is a proxy for allowing external services to do health checking

File: notm/prod/main.tf [hunk 12]
```diff
@@ -0,0 +1,218 @@
...
+    name = "webapp-config"
+    resource_group_name = "${azurerm_resource_group.group.name}"
+    deployment_mode = "Incremental"
+    template_body = "${file("../webapp-config.template.json")}"
+
+    parameters {
+        name = "${var.app-name}"
+        APPINSIGHTS_INSTRUMENTATIONKEY = "${azurerm_template_deployment.insights.outputs["instrumentationKey"]}"
+        NODE_ENV = "production"
+        API_ENDPOINT_URL = "https://noms-api-dev.dsd.io/elite2api/"
```
Comment (glenjamin) [inline notm/prod/main.tf]: should this be dev?

File: notm/prod/main.tf [hunk 13]
```diff
@@ -0,0 +1,218 @@
...
+    name = "webapp-ssl"
+    resource_group_name = "${azurerm_resource_group.group.name}"
+    deployment_mode = "Incremental"
+    template_body = "${file("../../shared/appservice-ssl.template.json")}"
+
+    parameters {
+        name = "${azurerm_template_deployment.webapp.parameters.name}"
+        hostname = "${azurerm_dns_cname_record.cname.name}.${azurerm_dns_cname_record.cname.zone_name}"
+        keyVaultId = "${azurerm_key_vault.vault.id}"
+        keyVaultCertName = "csraDOTserviceDOThmppsDOTdsdDOTio"
```
Comment (glenjamin) [inline notm/prod/main.tf]: needs updating

File: notm/prod/main.tf [hunk 14]
```diff
@@ -0,0 +1,218 @@
...
+        service = "${var.tags["Service"]}"
+        environment = "${var.tags["Environment"]}"
+    }
+
+    depends_on = ["azurerm_template_deployment.webapp"]
+}
+
+resource "azurerm_dns_cname_record" "cname" {
+    name = "notm"
+    zone_name = "hmpps.dsd.io"
```
Comment (glenjamin) [inline notm/prod/main.tf]: should be `service.hmpps.dsd.io`

**Comment by Mjwillis** [comment]
Yes true, but I thought I’d add the scripts anyway.  Do we have to run them? I assume we can just run pre-prod for now?

If not I’ll delete.

Mike



Michael Willis | Developer / TA
Syscon Justice Systems International Ltd.
HMPPS Digital Studio - 32 Scotland Street, Sheffield, S3 7DQ
Mobile (+44) 7771 654422
www.syscon.net | michael.willis@digital.justice.gov.uk

File: .gitignore (hunk 1 above)
Comment (Mjwillis) [inline .gitignore]: Ok will remove

**Comment by glenjamin** [comment]
> Yes true, but I thought I’d add the scripts anyway. Do we have to run them? I assume we can just run pre-prod for now?

The idea is that git reflects what's been provisioned, so would rather leave them out for now.

**Comment by Mjwillis** [comment]
@glenjamin made changes, only one I'm not sure of is the  keyVaultCertName = "notm-preprodDOTserviceDOThmppsDOTdsdDOTio"  - is that correct.  I assume we will have different certs for pre-prod and prod?

**Comment by glenjamin** [comment]
Merged via a rebase.

**Pull request closed without merging**